jwst/cube_build/blot_cube.py

```diff
@@ -0,0 +1,50 @@
+""" Map the detector pixels to the cube coordinate system.
+This is where the weight functions are used.
+"""
+import numpy as np
+import logging
+log = logging.getLogger(__name__)
+log.setLevel(logging.DEBUG)
+
+
+def blot_overlap(ipt, xstart,
+                 xcenter, ycenter,
+                 x_cube, y_cube,
+                 flux_cube,
+                 blot_xsize,
+                 blot_flux, blot_weight):
+
+    """ Blot the median sky image back to the detector
+
+    ipt is the median element.
+    xcenter, ycenter are the detector pixel arrays
+    xstart is only valid for MIRI and is the start of the x detector value for channel
+          0  for channels on left and ~512 for channels on right
+    x_cube, y_cube:  median cube IFU mapped backwards to detector
+    flux_cube: median flux
+    blot_flux & blot_weight: blotted values of flux_cube to detector
+    """
+
+    roi_det = 1.0  # Just large enough that we don't get holes
+    xdistance = np.absolute(x_cube[ipt] - xcenter)
+    ydistance = np.absolute(y_cube[ipt] - ycenter)
+
+    index_x = np.where(xdistance <= roi_det)
+    index_y = np.where(ydistance <= roi_det)
```

Comment:
I am going to hold off making changes to blot_cube.py because I have another JP ticket to work just on blot cube after I get the c extensions in this PR committed. I will come back to these changes suggestions later this week.